kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/throwable/matchers.kt

```diff
@@ -0,0 +1,63 @@
+package io.kotlintest.matchers.throwable
+
+import io.kotlintest.Matcher
+import io.kotlintest.Result
+import io.kotlintest.should
+import io.kotlintest.shouldNot
+
+infix fun Throwable.shouldHaveMessage(message: String) = this should haveMessage(message)
+infix fun Throwable.shouldNotHaveMessage(message: String) = this shouldNot haveMessage(message)
+fun haveMessage(message: String) = object : Matcher<Throwable> {
+  override fun test(value: Throwable) = Result(
+      value.message == message,
+      "Throwable should have message \"$message\", but instead got ${value.message}",
+      "Throwable should not have message \"$message\""
+  )
+}
+
+fun Throwable.shouldHaveCause(block: ((Throwable) -> Unit)? = null) {
```

Comment:
Perhaps it's better to use
```
block: (Throwable) -> Unit = {}
```

What do you think?

Reply:
done!